src/kinfitter.cc

```diff
@@ -0,0 +1,222 @@
+            std::cout<<"ECON MET"<<std::endl;
+            std::cout<<"ECON (E,Px,Py,Pz,M) "<<","<< ptmiss.Px() <<","<< ptmiss.Py() <<std::endl;
+            std::cout<<"ECON METCOV "<<std::endl;
+            std::cout<<"ECON "<< metcov (0,0) <<"  "<< metcov (0,1) << std::endl;
+            std::cout<<"ECON "<< metcov (1,0) <<"  "<< metcov (1,1) << std::endl;
+            std::cout<<"ECON tau1, tau2, b1, b2"<<std::endl;
+            std::cout<<"ECON ";
+            tlv_l1.Print();
+            std::cout<<"ECON ";
+            tlv_l2.Print();
+            std::cout<<"ECON ";
+            tlv_b1.Print();
+            std::cout<<"ECON ";
+            tlv_b2.Print();
+        }
+        wrongFit=true;
+    }
+    if(!wrongFit) {
+        localChi2 = localFit.getChi2(mh1_hp, mh2_hp);
+        if (localChi2 < 0 || std::isnan(localChi2)) {
+            localChi2 = std::nanf("1");
+        }
+    }
+    else {
+        localChi2 = std::nanf("1");
+    }
+
+    return std::pair(localFit, localChi2);
+}
+
+bool KinFitter::fit(std::string sgnHp) {
+    int mh1_hp = 0;
+    int mh2_hp = 0;
+
+    if (sgnHp == "HH")
+    {
+        mh1_hp = H_MASS;
+        mh2_hp = H_MASS;
+    }
+    else if (sgnHp == "ZZ") {
+        mh1_hp = Z_MASS;
+        mh2_hp = Z_MASS;
+    }
+    else if (sgnHp == "ZH") {
+        mh1_hp = Z_MASS;
+        mh2_hp = H_MASS;
+    }
+    else {
+        std::cout << "**ERROR: only HH/ZH/ZZ supported in KinFitter class, " << sgnHp << " entered." << std::endl;
+        std::cout << "         defaulting to HH hypothesis!" <<std::endl;
+        mh1_hp = H_MASS;
+        mh2_hp = H_MASS;
+    }
+
+    if (mh1_hp == mh2_hp) {
+        std::pair<HHKinFit2::HHKinFitMasterHeavyHiggs,float> res = _fit(mh1_hp, mh2_hp);
+        if (std::isnan(res.second)) fitsNotConverging = true;;
+        KinFit = res.first;
+        mh1_hpFit = mh1_hp;
+        mh2_hpFit = mh2_hp;
+    }
+    else {
+        std::pair<HHKinFit2::HHKinFitMasterHeavyHiggs,float> right_fit = _fit(mh1_hp, mh2_hp);
+        std::pair<HHKinFit2::HHKinFitMasterHeavyHiggs,float> left_fit  = _fit(mh2_hp, mh1_hp);
+
+        if (!std::isnan(right_fit.second) && !std::isnan(left_fit.second )) {
+            if (right_fit.second < left_fit.second) {
+                KinFit = right_fit.first;
+                mh1_hpFit = mh1_hp;
+                mh2_hpFit = mh2_hp;
+            }
+            else {
+                KinFit = left_fit.first;
+                mh1_hpFit = mh2_hp;
+                mh2_hpFit = mh1_hp;
+            }
+        }
+        else if (std::isnan(right_fit.second)) {
+            KinFit = left_fit.first;
+            mh1_hpFit = mh2_hp;
+            mh2_hpFit = mh1_hp;
+        }
+        else if (std::isnan(left_fit.second)) {
+            KinFit = right_fit.first;
+            mh1_hpFit = mh1_hp;
+            mh2_hpFit = mh2_hp;
+        }
+        else fitsNotConverging = true;;
+    }
+
+    return fitsNotConverging;
+}
+
+double KinFitter::getMH() {
```

Comment:
All these "get" function work only if one has already called `KinFitter::fit` in the code, otherwise i'm not sure what the return.